R/fetch-coaches-votes.R

```diff
@@ -0,0 +1,134 @@
+#'
+#' @return
+#' A data frame with columns: Season, Round, Finals, Home.Team, Away.Team, Player.Name, Coaches.Votes
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # Return all coaches votes across all seasons
+#' fetch_coaches_votes(comp = "AFLM", season = 2007:2021)
+#' fetch_coaches_votes(comp = "AFLW", season = 2018:2021)
+#'
+#' # Return all coaches votes for a particular round
+#' fetch_coaches_votes(comp = "AFLM", season = 2021, round = 24)
+#' fetch_coaches_votes(comp = "AFLW", season = 2021, round = 9)
+#'
+#' # Return all coaches votes for a particular team
+#' fetch_coaches_votes(comp = "AFLM", season = 2021, team = "Western Bulldogs")
+#' fetch_coaches_votes(comp = "AFLW", season = 2021, team = "Western Bulldogs)
+#'
+#' # Return all coaches votes for a particular match
+#' fetch_coaches_votes(comp = "AFLM", season = 2021, round = 24, team = "Western Bulldogs")
+#' fetch_coaches_votes(comp = "AFLW", season = 2021, round = 9, team = "Western Bulldogs)
+#' }
+#'
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+fetch_coaches_votes <- function(comp = "AFLM", season, round = 1:27,
+                                team = c("Adelaide Crows", "Brisbane Lions", "Carlton", "Collingwood",
+                                         "Essendon", "Fremantle", "Geelong Cats", "Gold Coast Suns",
+                                         "GWS Giants", "Hawthorn", "Melbourne", "North Melbourne",
+                                         "Port Adelaide", "Richmond", "St Kilda", "Sydney Swans",
+                                         "West Coast Eagles", "Western Bulldogs")){
+  
+  # packages
+  require(tidyverse)
+  
+  # error handling
+  if(!comp%in%c("AFLM","AFLW")) stop("Invalid Competition")
+  if(sum(!team%in%c("Adelaide Crows", "Brisbane Lions", "Carlton", "Collingwood", "Essendon", "Fremantle",
+                    "Geelong Cats", "Gold Coast Suns", "GWS Giants", "Hawthorn", "Melbourne", "North Melbourne",
+                    "Port Adelaide", "Richmond", "St Kilda", "Sydney Swans", "West Coast Eagles", "Western Bulldogs")
+         )>0) stop("Invalid team")
+  
+  all_coaches_votes <- expand.grid(Season = season, Round = round, Finals = c(F,T)) %>%
+    as.data.frame %>%
+    # exclude obvious impossibilities
+    subset(!(
+      (Season<2018 & Finals) |
+        (Round<19 & Finals) |
+        (Round>23 & !Finals)
+    )) %>%
+    split(1:nrow(.)) %>%
+    # apply function to each round
+    lapply(function(row){
+      try(scrape_coaches_votes(comp, row$Season, row$Round, row$Finals))
+    })
+  
+  # remove errors
+  all_coaches_votes[sapply(all_coaches_votes, typeof)=="character"] <- NULL
+  
+  if(length(all_coaches_votes) == 0) stop("No matches returned")
+  
+  # create data frame
+  all_coaches_votes <- do.call(rbind, all_coaches_votes) %>%
+    subset(Home.Team%in%team | Away.Team%in%team)
+  
+  return(all_coaches_votes)
+}
+#' @rdname fetch_coaches_votes
+#' @export
```

Comment:
It might be better to make this an internal function and not export it. See any of the `helpers` files to what this looks like but have included an example below


```
#' Check Season
#'
#' Checks the season for various things
#'
#' @param x Season in Year format
#'
#' @keywords internal
#' @noRd
check_season <- function(x) {
...
```

Reply:
Did you mean scrape_coaches_votes or both? And which helpers file should I put it in, or make a new one?

Reply:
Sorry for the late reply on this - didn't realise you had added a question. 

I think just for the `scrape_coaches_votes` you can make internal and add to a helper file. I would create a new helper file called `helper-aflcoaches.R`